File: python/7 kyu/279.py
# https://www.codewars.com/kata/5641275f07335295f10000d0/train/python

# Passed

def split_the_bill(x):
    average = sum(x.values()) / len(x)
    result = {
        person: round(spend_amt - average, 2) for person, spend_amt in x.items()
    }
    
    return result

output = split_the_bill({'A': 20, 'B': 15, 'C': 10})
print(output)

File: nodejs/5 kyu/279.js
// https://www.codewars.com/kata/559a28007caad2ac4e000083/train/javascript

// Passed

function perimeter(n) {
  let memo = {};
  let calculateFibo = (i) => {
    if (i === 0 || i === 1) {
      return 1;
    }

    if (i in memo) {
      return memo[i];
    }

    let result = calculateFibo(i - 1) + calculateFibo(i - 2);
    memo[i] = result;
    return result;
  };

  let perimeter = 0;
  for (let squareCount = 0; squareCount <= n; squareCount++) {
    let fiboNum = calculateFibo(squareCount);
    perimeter += fiboNum * 4;
  }

  return perimeter;
}

const output = perimeter(30);
console.log(output);